src/main/java/org/fcrepo/migration/f4clients/OCFLFedora4Client.java

```diff
@@ -15,28 +26,53 @@
 
     private static final Logger LOGGER = getLogger(OCFLFedora4Client.class);
 
+    private final OcflRepository ocflRepo;
+
+    private final String storageRoot;
+
+    private final String stagingRoot;
 
     /**
      * Constructor
-     * 
+     *
+     * @author Remigiusz Malessa
+     * @since 4.4.1-SNAPSHOT
      * @param storage Root for OCFL Objects
      * @param staging directory for in-progress OCFL Objects
      */
     public OCFLFedora4Client(final String storage, final String staging) {
-        LOGGER.info("to-be-implemented: OCFLFedora4Client: " + storage + "," + staging);
+
+        this.storageRoot = storage;
+        this.stagingRoot = staging;
+        // If locations exist instantiate repository
+        if (new File(storage).isDirectory() && new File(staging).isDirectory()) {
+            final Path repoDir = Paths.get(this.storageRoot);
+            Security.addProvider(new BouncyCastleProvider());
```

Comment:
What does this do?
```
Security.addProvider(new BouncyCastleProvider());
```

Reply:
I was just browsing the PR and saw this. That installs BouncyCastle as a security provider. It is only necessary if you intend to use blake2b, which JCE does not provide an implementation of, or if you don't like JCE.

Reply:
Removed BouncyCastleProvider.